source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -358,6 +363,19 @@ void DnsCacheImpl::finishResolve(const std::string& host,
                                  absl::optional<MonotonicTime> resolution_time,
                                  bool is_proxy_lookup) {
   ASSERT(main_thread_dispatcher_.isThreadSafe());
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.dns_cache_set_ip_version_to_remove")) {
+    {
+      absl::MutexLock lock{&ip_version_to_remove_lock_};
+      if (ip_version_to_remove_.has_value()) {
```

Comment:
if we're stuck with locks let's do a locked get of this and remove addresses outside of the lock

Reply:
Actually, the more I think about it, I think not holding the lock while deleting the IP address from the DNS response can cause a race condition although it's probably pretty rare. The test (in CI at least) is also failing if we hold a lock while removing the IP addresses.

Reply:
I'd like to understand this one - will ping